build-tools/build-indexes

```diff
@@ -1017,7 +1016,216 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
+			let tierOrder = ["OU", "AG", "Uber", "(Uber)", "OU", "(OU)", "UUBL", "UU", "RUBL", "RU", "NUBL", "NU", "PUBL", "PU", "(PU)", "New", "NFE", "LC Uber", "LC", "Unreleased"];
+			if (typeof modTiers !== "undefined") {
+				if (ClientMods[modid].excludeStandardTiers) tierOrder = modTiers;
+				else tierOrder = modTiers.concat(tierOrder);
+			}
+			for (const tier of tierOrder) {
+				if (tier in {OU:1, AG:1, Uber:1, UU:1, "(UU)":1, RU:1, NU:1, "(NU)":1, PU:1, "(PU)":1, NFE:1, LC:1, DOU:1, DUU:1, "(DUU)":1, New:1, Legal:1, Regular:1, "Restricted Legendary":1, "CAP LC":1}) {
+					let usedTier = tier;
+					if (usedTier === "(UU)") usedTier = "RU";
+					if (usedTier === "(NU)") usedTier = "PU";
+					if (usedTier === "(PU)") usedTier = "ZU";
+					if (usedTier === "(DUU)") usedTier = "DNU";
+					formatSlices[usedTier] = tiers.length;
+				}
+				if (!tierTable[tier]) continue;
+				tiers.push(['header', tier]);
+				tiers.push(...tierTable[tier]);
+			}
+			//pokemon stats
+			const overrideStats = {};
+			BattleTeambuilderTable[modid].overrideStats = overrideStats;
+			const overrideType = {};
+			BattleTeambuilderTable[modid].overrideType = overrideType;
+			const overrideAbility = {};
+			BattleTeambuilderTable[modid].overrideAbility = overrideAbility;
+			const overrideSecondAbility = {};
+			BattleTeambuilderTable[modid].overrideSecondAbility = overrideSecondAbility;
+			const overrideHiddenAbility = {};
+			BattleTeambuilderTable[modid].overrideHiddenAbility = overrideHiddenAbility;
+			const removeSecondAbility = {};
+			BattleTeambuilderTable[modid].removeSecondAbility = removeSecondAbility;
+			const requiredItem = {};
+			BattleTeambuilderTable[modid].requiredItem = requiredItem;
+			const fullFakemonData = {};
+			BattleTeambuilderTable[modid].fullFakemonData = fullFakemonData;
+			for (const id in modData.Pokedex) {
+				const modEntry = modData.Pokedex[id];
+				const baseEntry = Dex.data.Pokedex[id];
+				const baseType = () => {
+					if (typeof baseEntry !== 'undefined') return baseEntry.types.join('/');
+				};
+				const fakeMon = (typeof baseEntry === 'undefined');
+				if (fakeMon) {
+					fullFakemonData[id] = modEntry;
+				} else {
+					for (const stat in modEntry.baseStats) {
+						if (modEntry.baseStats[stat] !== baseEntry.baseStats[stat]) {
+							if (!overrideStats[id]) overrideStats[id] = {};
+							overrideStats[id][stat] = modEntry.baseStats[stat];
+						}
+					}
+					if (modEntry.types.join('/') !== baseType) {
+						overrideType[id] = modEntry.types.join('/');
+					}
+					if (modEntry.abilities['0'] !== baseEntry.abilities['0']) {
+						overrideAbility[id] = modEntry.abilities['0'];
+					}
+					if (modEntry.abilities['1'] !== baseEntry.abilities['1']) {
+						overrideSecondAbility[id] = modEntry.abilities['1'];
+					}
+					if (modEntry.abilities['H'] !== baseEntry.abilities['H']) {
+						overrideHiddenAbility[id] = modEntry.abilities['H'];
+					}
```

Comment:
```suggestion
					for (const ability of ['0', '1', 'H']) {
						if (modEntry.abilities[ability] !== baseEntry.abilities[ability]) {
							overrideAbility[id] = modEntry.abilities[ability];
						}
					}

Reply:
This doesn't work because overrideAbility only overrides the first ability. The pre-existing code only checks for overrideAbility and overrideHiddenAbility in BattleTeambuilderTable. overrideSecondAbility is something I added. If you don't think it should be done that way, I'll have to change the way past gen abilities are done as well (willing to, just let me know).